fvm/fvm_fuzz_test.go

```diff
@@ -0,0 +1,259 @@
+package fvm_test
+
+import (
+	"fmt"
+	"math"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/cadence"
+	jsoncdc "github.com/onflow/cadence/encoding/json"
+
+	"github.com/onflow/flow-go/engine/execution/testutil"
+	"github.com/onflow/flow-go/fvm"
+	"github.com/onflow/flow-go/fvm/errors"
+	"github.com/onflow/flow-go/fvm/meter"
+	"github.com/onflow/flow-go/fvm/programs"
+	"github.com/onflow/flow-go/fvm/state"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+func FuzzTransactionComputationLimit(f *testing.F) {
+	// setup initial state
+	vmt, tctx := bootstrapFuzzStateAndTxContext(f)
+
+	f.Add(uint64(0), uint64(0), uint(0))
+	f.Add(uint64(5), uint64(0), uint(0))
+	f.Fuzz(func(t *testing.T, computationLimit uint64, memoryLimit uint64, transactionType uint) {
+		computationLimit %= flow.DefaultMaxTransactionGasLimit
```

Comment:
why %=?  should be min?

Reply:
it is something like: `random(0, flow.DefaultMaxTransactionGasLimit)`

Reply:
I did some searching.

I do want randomness here. that is the whole point of the fuzzing tests. `%` preserves the randomness of whatever the fuzzing framework provides. Another option would be skipping if out of bounds. not sure I like that.